packages/zwave-js/src/lib/commandclass/DoorLockLoggingCC.ts

```diff
@@ -0,0 +1,662 @@
+		EventType.LockCodeOOSchedule,
+		EventType.UnlockCodeOOSchedule,
+		EventType.IllegalCode,
+		EventType.LockManual,
+		EventType.UnlockManual,
+		EventType.LockAuto,
+		EventType.UnlockAuto,
+		EventType.LockRemoteCode,
+		EventType.UnlockRemoteCode,
+		EventType.LockRemote,
+		EventType.UnlockRemote,
+		EventType.LockRemoteCodeOOSchedule,
+		EventType.UnlockRemoteCodeOOSchedule,
+		EventType.RemoteIllegalCode,
+		EventType.LockManual2,
+		EventType.UnlockManual2,
+		EventType.LockSecured,
+		EventType.LockUnsecured,
+		EventType.UserCodeAdded,
+		EventType.UserCodeDeleted,
+		EventType.AllUserCodesDeleted,
+		EventType.MasterCodeChanged,
+		EventType.UserCodeChanged,
+		EventType.LockReset,
+		EventType.ConfigurationChanged,
+		EventType.LowBattery,
+		EventType.NewBattery,
+		EventType.Unknown,
+	];
+
+	if (!valueDB.hasMetadata(recordEventTypeID)) {
+		valueDB.setMetadata(recordEventTypeID, {
+			...ValueMetadata.ReadOnlyString,
+			label: `record event type`,
+			description: "Event Type for latest logging record",
+			states: enumValuesToMetadataStates(EventType, supportedEventTypes),
+		});
+	}
+	if (
+		!valueDB.hasMetadata(recordNumberID) &&
+		typeof supportedRecordsNumber != "undefined"
+	) {
+		valueDB.setMetadata(recordNumberID, {
+			...ValueMetadata.ReadOnlyNumber,
+			min: 1,
+			max: supportedRecordsNumber,
+			label: `record number`,
+			description: "Record Number for latest logging record",
+		});
+	}
+	if (!valueDB.hasMetadata(recordUserIDID)) {
+		valueDB.setMetadata(recordUserIDID, {
+			...ValueMetadata.ReadOnlyNumber,
+			min: 0,
+			label: `record user id`,
+			description: "User ID for latest logging record",
+		});
+	}
+
+	const codeMetadata: ValueMetadata = {
+		...(Buffer.isBuffer(userCode)
+			? ValueMetadata.Buffer
+			: ValueMetadata.String),
+		minLength: 4,
+		maxLength: 10,
+		label: `record user code`,
+	};
+	if (valueDB.getMetadata(recordUserCodeID)?.type !== codeMetadata.type) {
+		valueDB.setMetadata(recordUserCodeID, codeMetadata);
+	}
+}
+
+@API(CommandClasses["Door Lock Logging"])
+export class DoorLockLoggingCCAPI extends PhysicalCCAPI {
+	public supportsCommand(cmd: DoorLockLoggingCommand): Maybe<boolean> {
+		switch (cmd) {
+			case DoorLockLoggingCommand.RecordsSupportedGet:
+			case DoorLockLoggingCommand.RecordsSupportedReport:
+			case DoorLockLoggingCommand.RecordGet:
+			case DoorLockLoggingCommand.RecordReport:
+				return true;
+		}
+		return super.supportsCommand(cmd);
+	}
+
+	protected [POLL_VALUE]: PollValueImplementation = async ({
+		property,
+	}): Promise<unknown> => {
+		switch (property) {
+			case "supportedRecordsNumber":
+				return (await this.getSupportedRecordsNumber())?.[property];
+			case "record":
+				// Fetches latest record by default
+				return (await this.getRecord(LATEST_RECORD_NUMBER_KEY))?.[
+					property
+				];
+			default:
+				throwUnsupportedProperty(this.ccId, property);
+		}
+	};
+
+	public async getSupportedRecordsNumber(): Promise<any> {
```

Comment:
Let's call this `getRecordCount` or something like that. The "official" Z-Wave commands often have weird names and I like to simplify them for the CCAPI methods.